File: BE/src/main/java/com/all_i/allibe/post/domain/MealTime.java
package com.all_i.allibe.post.domain;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor(access = AccessLevel.PROTECTED)
public enum MealTime {
    MORNING("아침"),
    LUNCH("점심"),
    DINNER("저녁");

    private final String key;
}

File: FE/src/types/MealTime.ts
export enum MealTime {
    MORNING = "아침",
    LUNCH = "점심",
    DINNER = "저녁",
    NONE = "알 수 없음",
}

const MealTimeEnglishMap: Record<MealTime, string> = {
    [MealTime.MORNING]: "MORNING",
    [MealTime.LUNCH]: "LUNCH",
    [MealTime.DINNER]: "DINNER",
    [MealTime.NONE]: "UNKNOWN",
};

export function getMealTime(): string {
    const currentHour = new Date().getHours();

    let mealTime: MealTime;
    if (currentHour >= 4 && currentHour < 10) {
        mealTime = MealTime.MORNING;
    } else if (currentHour >= 10 && currentHour < 16) {
        mealTime = MealTime.LUNCH;
    } else {
        mealTime = MealTime.DINNER;
    }

    return MealTimeEnglishMap[mealTime];
}
